extensions/target-specific/chromium/compositorframe/compositorframe.js

```diff
@@ -0,0 +1,102 @@
+//--------------------------------------------------------------
+//
+//    MIT License
+//
+//    Copyright (c) Microsoft Corporation. All rights reserved.
+//
+//--------------------------------------------------------------
+
+"use strict";
+
+var CompositorFrame = undefined;
+Loader.OnLoad(function () {
+  CompositorFrame = {
+    Tree: new DbgObjectTree.DbgObjectTreeReader(),
+    Renderer: new DbgObjectTree.DbgObjectRenderer(),
+    InterpretAddress: function (address) {
+      var voidObj = DbgObject.create("ntdll!void", address);
+      if (!voidObj.isNull()) {
+        return voidObj.as(Chromium.GpuProcessType("viz::CompositorFrame"));
+      }
+      return DbgObject.NULL;
+    },
+    GetRoots: function () {
+      return DbgObject.locals("service", "viz::CompositorFrameSinkImpl::SubmitCompositorFrameInternal", "frame").then(
+        (frames) => {
+          return frames;
+        }
+      );
+    },
+    DefaultTypes: []
+  };
+
+  CompositorFrame.Tree.addChildren(Chromium.GpuProcessType("viz::CompositorFrame"), (parentLayer) => {
+    return parentLayer.f("render_pass_list").array("Elements").map(unique_ptr => { return unique_ptr.F("Object"); });
```

Comment:
You can make **unique_ptr => { return unique_ptr.F("Object"); }** even cleaner, like this: **unique_ptr => unique_ptr.F("Object")**